extensions/arc/src/ui/dashboards/postgres/postgresResourceHealthPage.ts

```diff
@@ -0,0 +1,343 @@
+			label: loc.refresh,
+			iconPath: IconPathHelper.refresh
+		}).component();
+
+		this.disposables.push(
+			refreshButton.onDidClick(async () => {
+				refreshButton.enabled = false;
+				try {
+					this.podConditionsLoading!.loading = true;
+
+					await Promise.all([
+						this._postgresModel.refresh(),
+						this._controllerModel.refresh()
+					]);
+				} catch (error) {
+					vscode.window.showErrorMessage(loc.refreshFailed(error));
+				}
+				finally {
+					refreshButton.enabled = true;
+				}
+			}));
+
+		return this.modelView.modelBuilder.toolbarContainer().withToolbarItems([
+			{ component: refreshButton, toolbarSeparatorAfter: true }
+		]).component();
+	}
+
+	private createPodList(): string[] {
+		const podStatus = this._postgresModel.config?.status.podsStatus;
+		let podNames: string[] = [];
+
+		podStatus?.forEach(p => {
+			let podHealthModels: PodHealthModel[] = [];
+			let indexes: number[] = [];
+
+
+			p.conditions.forEach(c => {
+				const condtionContainer = this.modelView.modelBuilder.flexContainer().withProps({
+					CSSStyles: { 'alignItems': 'center', 'height': '15px' }
+				}).component();
+
+				if (c.status === 'False') {
+					const imageComponent = this.modelView.modelBuilder.image().withProps({
+						iconPath: IconPathHelper.fail,
+						width: iconSize,
+						height: iconSize,
+						iconHeight: '15px',
+						iconWidth: '15px'
+					}).component();
+					condtionContainer.addItem(imageComponent, { CSSStyles: { 'margin-right': '0px' } });
+
+					condtionContainer.addItem(this.modelView.modelBuilder.text().withProps({
+						value: c.message,
+					}).component());
+				} else {
+					const imageComponent = this.modelView.modelBuilder.image().withProps({
+						iconPath: IconPathHelper.success,
+						width: iconSize,
+						height: iconSize,
+						iconHeight: '15px',
+						iconWidth: '15px'
+					}).component();
+					condtionContainer.addItem(imageComponent, { CSSStyles: { 'margin-right': '0px' } });
+
+					if (c.type === 'Initialized') {
+						condtionContainer.addItem(this.modelView.modelBuilder.text().withProps({
+							value: loc.podInitialized,
+						}).component());
+					} else if (c.type === 'Ready') {
+						condtionContainer.addItem(this.modelView.modelBuilder.text().withProps({
+							value: loc.podReady,
+						}).component());
+					} else if (c.type === 'ContainersReady') {
+						condtionContainer.addItem(this.modelView.modelBuilder.text().withProps({
+							value: loc.containerReady,
+						}).component());
+					} else if (c.type === 'PodScheduled') {
+						condtionContainer.addItem(this.modelView.modelBuilder.text().withProps({
+							value: loc.podScheduled,
+						}).component());
+					}
+				}
+
+				indexes.push(this.podsData.length);
+				this.podsData.push({
+					condition: c.type,
+					details: condtionContainer,
+					lastUpdate: c.lastTransitionTime
+				});
+			});
+
+			if (p.role.toUpperCase() !== loc.coordinator.toUpperCase()) {
```

Comment:
What's the reason for treating the coordinator differently?  Is it because we want the coordinator to be the default pod in the drop down?

Reply:
That is the reason I treat it differently.